### ISSUE TITLE
Text getting garbled when rendering 2+ PDFs on same page

### ISSUE THREAD

**Issue by jakeg** [mozilla/pdf.js#15876]
Attach (recommended) or Link to PDF file here:

[example1.pdf](https://github.com/mozilla/pdf.js/files/10315857/example1.pdf)
[example2.pdf](https://github.com/mozilla/pdf.js/files/10315856/example2.pdf)

Configuration:
- Web browser and its version: any (happens in both Chrome or Firefox)
- Operating system and its version: I presume any (tested on Windows 11)
- PDF.js version: 3.1.81
- Is a browser extension: no

Steps to reproduce the problem:
1. Open https://www.mathsuniverse.com/infinity/pdf-test
2. Drag the 1st PDF in, then scroll right with the keyboard right arrow a bit, then drag the 2nd PDF in
3. Force re-render of 1st PDF at a new zoom level by zooming in with + key or mouse scroll wheel

What is the expected behavior? Both PDFs to show correctly as shown

![image](https://user-images.githubusercontent.com/744792/209858155-fd149498-129b-4d0c-bd80-292f5abbf01a.png)

What went wrong? Text in the 1st PDF to be imported gets garbled as shown

![image](https://user-images.githubusercontent.com/744792/209858294-180a01cb-b761-41c5-81b9-2b20250b7172.png)

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension): https://www.mathsuniverse.com/infinity/pdf-test

--

I presume the problem is that PDF.js (edit: PDFLib was a typo) isn't expecting to be working with multiple PDFs (perhaps with different text encoding or something) at once? Or every time I switch to rendering a page in a different PDF, is there some API I need to call first to load up the right encoding or fonts or something? My PDFjs-specific code isn't doing anything much more than in the [Hello world example](https://github.com/mozilla/pdf.js/blob/master/examples/learning/helloworld.html).

**Comment by Snuffleupagus** [comment]
> I presume the problem is that PDFLib isn't expecting to be working with multiple PDFs (perhaps with different text encoding or something) at once?

Sorry, but we don't know what "PDFLib" is since that's not something that's being developed here and consequently we cannot provide any help/support for that unfortunately. Does it work if you don't use "PDFLib", since it's certainly possible that it's not compatible with the PDF.js library?

> Or every time I switch to rendering a page in a different PDF, is there some API I need to call first to load up the right encoding or fonts or something?

There's logic in the PDF.js library that should ensure that different documents get unique font-identifiers, specifically to prevent issues with multiple documents being opened in parallel. (However, we often recommend using `iframe`s to fully isolate documents from each other.)

> My PDFjs-specific code isn't doing anything much more than in the [Hello world example](https://github.com/mozilla/pdf.js/blob/master/examples/learning/helloworld.html).

Your example linked above unfortunately looks too large/complex, please see https://github.com/mozilla/pdf.js/blob/master/.github/CONTRIBUTING.md (emphasis mine):
> If you are developing a custom solution, first check the examples at https://github.com/mozilla/pdf.js#learning and search existing issues. If this does not help, **please prepare a short well-documented example that demonstrates the problem and make it accessible online** on your website, JS Bin, GitHub, etc. before opening a new issue or contacting us in the Matrix room -- **keep in mind that just code snippets won't help us troubleshoot the problem**.

**Comment by jakeg** [comment]
Apologies, I used to use PDFLib many years ago with PHP. That was a typo - I mean PDF.js.

```js
// When I import a PDF I do this...
const pdfs.push(await pdfjsLib.getDocument(url))

// Because I'm working on multiple PDFs, I imagine I need an extra line here to tell PDFjs to use the encoding/whatever for the correct PDF?

// Whenever re-rendering a page, eg when zoom level changes I do something like this
const page = await pdfs[i].getPage(1)
const viewport = page.getViewport({ scale: zoomLevel })
const renderTask = page.render({ canvasContext: ctx, viewport: viewport })
```

I'll try to make a pastebin for it as well.

**Comment by jakeg** [comment]
Bug was my end... each time I was loading a new PDF I was re-adding the PDF.js script. All fixed now, thanks for your quick reply.
